mozjs/js/rust/magic_codegen/src/lib.rs

```diff
@@ -35,18 +37,32 @@ fn match_struct(ast: &syn::DeriveInput, variant: &syn::VariantData) -> quote::To
         .split('_')
         .next()
         .expect("Should have a '_' in the magic dom struct name");
+    let dom_flag = if name.starts_with("DOM") {
+        quote::Ident::from("jsapi::JSCLASS_IS_DOMJSCLASS")
```

Comment:
I'm not sure it's necessary to set this flag. I just see the name and seems it may imply I need to add it.